bindings/java/src/test/com/apple/foundationdb/test/AsyncStackTester.java

```diff
@@ -509,6 +509,37 @@ else if (op == StackOperation.TENANT_LIST_NAMES) {
 				inst.push(output);
 			}, FDB.DEFAULT_EXECUTOR);
 		}
+		else if (op == StackOperation.TENANT_LIST_METADATA) {
```

Comment:
I hadn't considered this approach of validating the value as part of the instruction, but it seems sensible. If you are going to do this, my suggestion would be to leave it as one instruction (`TENANT_LIST`) and to convert the validation to some sort of assert that would cause the test to fail. Then you can continue storing just the names.